### ISSUE TITLE
FavoriteButton improvements

### ISSUE THREAD

**Issue by fstanis** [google/horologist#688]
- Similar to #677, we should make it a `ToggleButton`
- It should be disabled when nothing is playable (i.e. nothing to favorite)
- It needs to be aware of the current media (even if we don't implement persistence), otherwise the underlying mechanism is hard to implement; in practice we need to either:
  - Pass `MediaUiModel` or just an `id: String` to it
  - Extend `PlayerUiState` to contain arbitrary extras by adding a `Bundle` (or `Bundle`-like) object to it

**Comment by yschimke** [comment]
Closing as out of scope, since it's sample code.  There are bigger changes to make here.
